Moves some ghost role checks to client/

Rukofamicom opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
closes #8496 

Shuffles a set of checks out of `/obj/effect/mob_spawn/attack_ghost(mob/user)` 
and places it into a new proc: `/client/proc/canGhostRole(role, use_cooldown = FALSE, atom/callsource)

Additionally applies this check to:
* Spider eggs
* Carp Portals
* All instances where set_playable() is called

Adds a variable for role directly to living defines, for use with antagonistic mobs that are set_playable()
* Slimes of all varieties have ROLE_SENTIENCE, including pyroclastic
* Spiders all have ROLE_SPIDER
* Assassination targets for explorers have ROLE_TRAITOR because it is the most relevant role available and adding an entirely new ban type specifically for assassination targets doesn't seem worthwhile for this mob. 

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Ghost roles should properly obey bans, server settings and admin toggles. 

Tying the relocated checks to the ghost client trying to take a spawn instead of individual spawners makes it **much** easier to implement and maintain proper checks on ghost role spawns. This is also why roles are assigned directly to relevant mobs in this PR. 

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.
-->

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>

There is very little I can show to prove that what I have done works. 
![image](https://user-images.githubusercontent.com/9547572/218608347-10a089d0-3550-4ff9-9d92-dcd0638702dc.png)

Mass sentience verb used (set_playable() called on all station mobs)
![image](https://user-images.githubusercontent.com/9547572/218618964-92e6eb13-8247-4043-99ef-1221adff5a33.png)

Tested with a few spawners and spiders with the debugger set to breakpoint on various checks, but since I have no database I can't 100% verify that this works as it appears to and/or should.
</details>

## Changelog
:cl:
code: Checks for role bans, ghost role toggles and ghost role cooldowns have been moved from spawners to client code for easier use in determining ghost role eligibility
fix: Spider eggs, carp portals and other playable mobs now respect role bans, ghost role toggles and ghost role cooldowns. 
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Rukofamicom committed:
Moves ghostrole eligibility to mob/

Checks if round has started
Checks if spawners have been disabled
Checks if player is banned from the role
Checks if player is on ghost role cooldown

Rukofamicom committed:
implements new proc for carp portal

Rukofamicom committed:
implements new proc for various set_playable() mobs

PowerfulBacon commented on code/_onclick/click.dm:
```diff
@@ -495,3 +495,17 @@
 
 	return FALSE
+
+/mob/proc/canGhostRole(role, use_cooldown = FALSE)
```
This should be a client proc actually since it only uses values from client

Rukofamicom replied:
:+1:

Will get it in a few hours

Rukofamicom replied:
Done, updated title, description and briefly re-tested to ensure checks were being performed correctly. 

I had to slightly change the proc so that whatever is being treated as a spawner can pass along its flags_1 variable.

Rukofamicom committed:
move canGhostRole from mob/ to client/

Rukofamicom committed:
optimize

no reason to pass the entire src when I only need one variable, not sure why I thought that was a good idea.
